### PR TITLE
[POC] standardize setup errors

### PR DESCRIPTION
**Pull request by karimJWP**
### This PR will...
not build. It is a POC in which all errors surfaced to viewer are converted to JWErrors.
### Why is this Pull Request needed?
This is a POC to verify the usefulness of the prework done for the Error Standardization epic and uncover flaws.

### Are there any Pull Requests open in other repos which need to be merged with this?
https://github.com/jwplayer/jwplayer-errors/pull/3
#### Addresses Issue(s):
JW8-1264

### CODE REVIEW

**Comment by johnBartos** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Danger failed to run `dangerfile.js`.
  </td>
    </tr>
  </tbody>
</table>



## Error ReferenceError
```
error is not defined
ReferenceError: error is not defined
    at checkExactPackageVersion (dangerfile.js:37:9)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:160:7)
```
### Dangerfile
```
32| });
33| 
34| async function checkExactPackageVersion() {
35|     const diff = await danger.git.diffForFile('package.json');
36|     if (diff && diff.added && diff.added.match(/([\^~])/)) {
-----------^
37|         error(`🥕 Only save exact versions of a dependency, without a ~ or ^.`);
38|     }
39| }
40| 
```
  
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/js/api/core-loader.js [hunk 1]
```diff
@@ -17,7 +18,7 @@ export default function loadCoreBundle(model) {
 
 export function chunkLoadErrorHandler(/* error */) {
     // Webpack require.ensure error: "Loading chunk 3 failed"
-    throw new Error('Network error');
+    throw SetupError(Code.NETWORK_ERROR);
```
Comment (robwalch) [inline src/js/api/core-loader.js]: You mentioned extending Error, but this is not a class.

File: src/js/program/program-controller.js [hunk 2]
```diff
@@ -367,7 +368,7 @@ class ProgramController extends Eventable {
                 // The provider we need couldn't be loaded
                 if (!ProviderConstructor) {
                     this._destroyActiveMedia();
-                    throw Error(`Failed to load media`);
```
Comment (robwalch) [inline src/js/program/program-controller.js]: Here `Error` is an error. It should be `new Error`.

File: src/js/api/core-loader.js (hunk 1 above)
Comment (karimJWP) [reply src/js/api/core-loader.js]: I have a separate PR for that https://github.com/jwplayer/jwplayer-errors/pull/4

**Pull request closed without merging**